### ISSUE TITLE
Question about event handling with IPRoute API

### ISSUE THREAD

**Issue by tobiasjakobi-compleo** [svinota/pyroute2#876]
Hello,

first of all I apologize for using the bugtracker as a means to get information. Didn't want to mail Peter directly, because maybe someone else might have a good hint for me.

Anyway, I'm trying to use v0.5.18 of pyroute2 to set routing rules (in the policy based routing sense) when the configuration of an interface changes. By change I mean adding of an IP address, update, remove, etc.

I'm pretty sure that the IPRoute API/class provides everything that is necessary to set the routing rules, but I'm unsure about how to implement the monitoring loop. I checked the examples, and `ip_monitor.py` looks like the kind of thing I want. OK, there's no looping here, but I guess that would be trivial. However the example uses a IPRSocket object, and as far as I understand, if I'm using that one, that I have to do pretty much everything myself.

So basically what I'm looking for is something like:
```
with IPRoute() as ipr:
    while ipr.wait_for_event():
        evt = ipr.get_event()
        manage_my_routing_rules(evt)
```

So ipr actually has a `get()` method, but apparantly it doesn't do what I want. Also there's `register_callback()`, which could be potentially be interesting. But I don't get my registered callback function to fire.

I guess I'm pretty much lost here. I'm grateful for any kind of pointers. Maybe there's an example of a one of the tests that is closer to what I want to achieve.

Thanks in advance!

With best wishes,
Tobias

**Comment by crosser** [comment]
I get `nlsock = IPRoute()`, obtain its file descriptor `nlfd = nlsock.fileno()`, and when `poll()` notifies about a read event on the descriptor, run `nlsock.get()` (in a loop) to read all messages.

**Comment by tobiasjakobi-compleo** [comment]
Hello Eugene,

but isn't that pretty much functionally equivalent to just doing this?

```
with IPRoute() as ipr:
    while True:
        ipr.get()
```

I'm asking because this doesn't work for me. While running this, `ip monitor` shows some netlink events occuring. But I don't get any output from the script.
